pkg/webui/console/views/device-general-settings/index.js

```diff
@@ -187,9 +200,12 @@ export default class DeviceGeneralSettings extends React.Component {
     // 2. Disable the section if the device is OTAA and joined since no fields are stored in the AS.
     // 3. Disable the section if NS is not in cluster, since activation mode is unknown.
     // 4. Disable the seciton if `application_server_address` is not equal to the cluster AS address.
+    // 5. Disable the section if the user has no appropriate rights to edit keys.
     const sameAsAddress = getComponentBaseUrl(asConfig) === device.application_server_address
     const isJoined = isDeviceJoined(device)
-    const asDisabled = !asEnabled || (isOTAA && !isJoined) || !nsEnabled || !sameAsAddress
+    const showKeys = mayReadKeys && mayEditKeys
+    const asDisabled =
```

Comment:
I think it's confusing if we disable the entire AS section only because someone doesn't have access to keys.

Reply:
Well, there are no fields other than the keys that require the `RIGHT_APPLICATION_DEVICES_WRITE_KEYS` right.

Reply:
Not right now, but in the future I can imagine we'll add other AS fields, perhaps the `skip_payload_crypto` field